packages/z2s/src/compiler.pg-test.ts

```diff
@@ -215,118 +234,483 @@ function createdAtToMillis(row: Record<string, unknown>) {
+      expect(zqlResult).toMatchInlineSnapshot(`
+                        [
+                          {
+                            "closed": true,
+                            "createdAt": 1743127752952,
+                            "description": "Description for issue 1",
+                            "id": "issue1",
+                            "labels": [
+                              {
+                                "id": "label1",
+                                "name": "Label 1",
+                              },
+                              {
+                                "id": "label2",
+                                "name": "Label 2",
+                              },
+                            ],
+                            "ownerId": null,
+                            "title": "Test Issue 1",
+                          },
+                          {
+                            "closed": false,
+                            "createdAt": 1743041352952,
+                            "description": "Description for issue 2",
+                            "id": "issue2",
+                            "labels": [
+                              {
+                                "id": "label2",
+                                "name": "Label 2",
+                              },
+                            ],
+                            "ownerId": "user1",
+                            "title": "Test Issue 2",
+                          },
+                          {
+                            "closed": true,
+                            "createdAt": 1742954952952,
+                            "description": "Description for issue 3",
+                            "id": "issue3",
+                            "labels": [
+                              {
+                                "id": "label1",
+                                "name": "Label 1",
+                              },
+                            ],
+                            "ownerId": "user2",
+                            "title": "Test Issue 3",
+                          },
+                        ]
+                      `);
+    });
 
-  test('complex query combining multiple features', async () => {
-    const query = issueQuery
-      .where('closed', '=', false)
-      .whereExists('labels', q => q.where('name', 'IN', ['Label 1', 'Label 2']))
-      .related('owner')
-      .related('comments', q =>
-        q.orderBy('createdAt', 'desc').limit(3).related('author'),
-      )
-      .orderBy('title', 'asc');
-    const sqlQuery = formatPg(compile(ast(query), schema.tables, query.format));
-    const pgResult = extractZqlResult(
-      await pg.unsafe(sqlQuery.text, sqlQuery.values as JSONValue[]),
-    );
-    expect(
-      mapResultToClientNames(await query.run(), schema, 'issue'),
-    ).toEqualPg(pgResult);
-  });
+    test('nested related with where clauses', async () => {
+      // TODO Change to filter comments on createdAt once
```

Comment:
@tantaman Filtering by createdAt didn't work on node-postgres, got an error that 1000 was an invalid type for the input.  This should be fixed by your upcoming change to pass args as text and explicitly convert them.